kratos/python_scripts/auxiliary_solver_function_utilities.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+import KratosMultiphysics
+
+def add_auxiliary_variables(model_part, aux_variable_list):
+  for i in range(aux_variable_list.size()):
+      variable_name = aux_variable_list[i].GetString()
+      variable = KratosMultiphysics.KratosGlobals.GetVariable(variable_name)
+      model_part.AddNodalSolutionStepVariable(variable)
+
+def add_auxiliary_dofs(model_part, aux_dofs_list, aux_reaction_list):
+  if (aux_dofs_list.size() != aux_reaction_list.size()):
+          raise Exception("DoFs list and reaction list should be the same size")
+  for i in range(aux_dofs_list.size()):
+      dof_variable_name = aux_dofs_list[i].GetString()
+      reaction_variable_name = aux_reaction_list[i].GetString()
+      if (KratosMultiphysics.KratosGlobals.HasVariable(dof_variable_name)):
+          if(KratosMultiphysics.KratosGlobals.GetVariableType(dof_variable_name) == "Double"): # Double variable
```

Comment:
I would also say that this fct should check that the var-types are consistent

Reply:
Do you mean consistent between the reaction and the dof?

Reply:
yes